assets/javascripts/needleeditor.js

```diff
@@ -46,12 +46,12 @@ NeedleEditor.prototype.init = function() {
 
   // double click switched type
   cv.canvas.addEventListener('dblclick', function(e) {
-    var idx = cv.get_selection_idx();
-    if (idx == -1) {
+    var selection = editor.selection();
+    var shape = selection.shape;
+    var a = selection.area;
+    if (!selection || !a) {
```

Comment:
I suppose it's not harmful but I'd rather test for `selection` before assigning `selection.area` to a variable. If we ever get static analysis I'd expect this to be flagged as well.

Reply:
Yes, it should have been checking for `!shape`. Fixed it.